flags/eigendaflags/cli.go

```diff
@@ -163,10 +181,96 @@ func CLIFlags(envPrefix, category string) []cli.Flag {
 			EnvVars:  []string{withEnvPrefix(envPrefix, "PUT_RETRIES")},
 			Category: category,
 		},
+	},
+		v2Flags(envPrefix, category)...,
+	)
+}
+
+func v2Flags(envPrefix, category string) []cli.Flag {
+	return []cli.Flag{
+		// EigenDA V2 specific flags //
+		&cli.BoolFlag{
+			Name:     V2Enabled,
+			Usage:    "Enable blob dispersal and retrieval against EigenDA v2 protocol",
+			EnvVars:  []string{withEnvPrefix(envPrefix, "V2_ENABLED")},
+			Required: false,
+		},
+		&cli.StringFlag{
+			Name:     CertVerifierAddrName,
+			Usage:    "Address of the EigenDABlobVerifier contract. Required for performing eth_calls to verify EigenDA certificates.",
+			EnvVars:  []string{withEnvPrefix(envPrefix, "CERT_VERIFIER_ADDR")},
+			Category: category,
+			Required: false,
+		},
+		&cli.DurationFlag{
+			Name:     RelayTimeoutName,
+			Usage:    "Timeout used when querying a relay for blob contents.",
+			EnvVars:  []string{withEnvPrefix(envPrefix, "RELAY_TIMEOUT")},
+			Value:    10 * time.Second,
+			Required: false,
+		},
+		&cli.DurationFlag{
+			Name:     ContractCallTimeoutName,
+			Usage:    "Timeout used when performing smart contract eth_calls",
+			EnvVars:  []string{withEnvPrefix(envPrefix, "CONTRACT_CALL_TIMEOUT")},
+			Value:    10 * time.Second,
+			Required: false,
+		},
+		&cli.UintFlag{
+			Name:     BlobVersionName,
+			Usage:    "Blob version used when dispersing. Currently only supports (0)",
+			EnvVars:  []string{withEnvPrefix(envPrefix, "BLOB_VERSION")},
+			Value:    uint(0),
+			Required: false,
+		},
+	}
+}
+
+func readPayloadClientConfig(ctx *cli.Context) v2_clients.PayloadClientConfig {
+	noPolynomial := ctx.Bool(DisablePointVerificationModeFlagName)
+	polyMode := codecs.PolynomialFormCoeff
+
+	// if point verification mode is disabled then blob is treated as evaluations and
```

Comment:
I think you have this reversed.

`PolynomialFormEval` is interpreted like this:  "The encoded payload is interpreted as a polynomial in eval form. Blobs must be in coeff form, therefore we must take the IFFT to turn our encoded payload into the correct form".

And the other side of the coin, `PolynomialFormCoeff` is interpreted like this: "The encoded payload is interpreted as a polynomial in coeff form. Blobs must be in coeff form, therefore we don't need to do any IFFT encoding, the bytes can be sent as-is"

Originally I had understood the meaning of these terms as you wrote here, but talking with @bxue-l2 and @samlaf, we came to an agreement on the descriptions I wrote above. I'm still not super happy with our terminology here, but a perfect solution has proven elusive.

Reply:
oof a bit inversal ngl - can update accordingly